server/series/api.py

```diff
@@ -0,0 +1,436 @@
+
+    series.teams.remove(team)
+
+    return 200, series
+
+
+@router.get("/{series_slug}/team/{team_slug}", response={200: TeamSchema, 400: Response})
+def get_series_team(
+    request: AuthenticatedHttpRequest, series_slug: str, team_slug: str
+) -> tuple[int, Team | message_response]:
+    try:
+        series = Series.objects.get(slug=series_slug)
+    except Series.DoesNotExist:
+        return 400, {"message": "Series does not exist"}
+
+    try:
+        team = Team.objects.get(slug=team_slug)
+    except Team.DoesNotExist:
+        return 400, {"message": "Team does not exist"}
+
+    if team not in series.teams.all():
+        return 400, {"message": f"{team.name} is not registered for {series.name} !"}
+
+    return 200, team
+
+
+@router.post(
+    "/{series_id}/team/{team_id}/invitation",
+    response={200: SeriesRosterInvitationCreateSchema, 400: Response, 401: Response},
+)
+def send_series_invitation(
+    request: AuthenticatedHttpRequest,
+    series_id: int,
+    team_id: int,
+    invitation_details: SeriesRosterInvitationCreateSchema,
+) -> tuple[int, SeriesRosterInvitation | message_response]:
+    try:
+        series = Series.objects.get(id=series_id)
+        team = Team.objects.get(id=team_id)
+    except (Series.DoesNotExist, Team.DoesNotExist):
+        return 400, {"message": "Series does not exist"}
+
+    if request.user not in team.admins.all():
+        return 401, {"message": "Only team admins can invite players for a series"}
+
+    if team not in series.teams.all():
+        return 400, {"message": f"{team.name} is not registered for {series.name}"}
+
+    try:
+        to_player = Player.objects.get(id=invitation_details.to_player_id)
+    except Player.DoesNotExist:
+        return 400, {"message": "Player does not exist"}
+
+    invitation = SeriesRosterInvitation(
+        series=series, from_user=request.user, to_player=to_player, team=team
+    )
+
+    if invitation_details.expires_on is not None:
+        invitation.expires_on = invitation_details.expires_on
+
+    invitation.save()
+    return 200, invitation
+
+
+@router.delete(
+    "/invitation/{invitation_id}",
+    response={200: Response, 400: Response, 401: Response},
+)
+def revoke_series_invitation(
+    request: AuthenticatedHttpRequest, invitation_id: int
+) -> tuple[int, message_response]:
+    try:
+        invitation = SeriesRosterInvitation(id=invitation_id)
+    except SeriesRosterInvitation.DoesNotExist:
+        return 400, {"messsage": "Series/Invitation does not exist"}
+
+    if request.user != invitation.from_user:
+        return 401, {
+            "message": "You cannot revoke this invitation since this invitation wasn't sent by you"
+        }
+
+    if invitation.status != SeriesRosterInvitation.Status.PENDING:
+        return 400, {
+            "message": f"You cannot revoke an invitation that has been {invitation.status}"
+        }
+
+    invitation.delete()
```

Comment:
nit: instead of deleting can we have another status that says REVOKED?

Reply:
done